src/persist-client/src/cache.rs

```diff
@@ -288,6 +316,29 @@ where
     fn as_any(self: Arc<Self>) -> Arc<dyn Any + Send + Sync> {
         self
     }
+
+    fn push_diff(&self, diff: VersionedData) {
```

Comment:
This function looks interesting to test, seems uncovered

Reply:
ooh I like this code coverage report!

one tricky thing in this case is that we're wrapping this feature behind a flag that defaults off so it's safer to ship / test on individual staging environments. if we ran the tests with `persist_pubsub_client_enabled: true` I'd expect this section to be covered